Clean up sub_6960AB function (peep_should_go_on_ride)

clamburger opened the pull request:
These changes primarily cleans up the sub_6960AB function, which is used whenever a peep needs to decide whether or not they should go on a ride or visit a shop.

My C is a little rusty, so let me know if something needs to be changed or if either of the changes to original behaviour should be reversed.

Bug fixes made in the process:
* If it was raining and a peep goes on a ride that's sufficiently undercover, the peep would bypass all excitement and intensity checks (original game bug)
* The "Ride 1 is really good value" thought was never able to be triggered due to an incorrect flag check.
* The "Ride 1 is really good value" thought is no longer triggered if the peep paid for park entry. Previously, it checked if the park was currently free, even if it wasn't when the peep entered. Now the "good value" logic matches the "bad value" logic. (original game bug)
* If a peep encountered a queue on their way to a different ride or shop, it would reset their destination even if they decided not to enter.
* The min and max intensity values in peep_on_enter_ride were reversed.
* Ride queue lengths are now set to zero when the "remove all guests" cheat is used

Other things identified:
* `ride->first_peep_in_queue` seems to refer to the *latest* peep to enter the queue, but I wasn't sufficiently confident to change to the name. 
* `ride->var_114` seems to indicate how much of the ride is undercover (this affects if peeps will go on the ride while it's raining). I wasn't able to figure out exactly what contributed to it (being underground is one thing)

duncanspumpkin commented on src/peep/peep.c:
```diff
@@ -7882,194 +7886,230 @@ static void peep_reset_ride_heading(rct_peep *peep)
+				else {
+					// Check if there's room in the queue for the peep to enter.
+					if (ride->first_peep_in_queue[entranceNum] != 0xFFFF) {
+						rct_peep *firstPeepInQueue = &(g_sprite_list[ride->first_peep_in_queue[entranceNum]].peep);
```
Use GET_PEEP macro for this.

clamburger replied:
Thanks, have made the change

duncanspumpkin commented:
Excellent work. The original bugs you fixed I agree that they sound like sensible changes. Var_114 I came across before and I can't remember why it wasn't named. I have a feeling it might be named in the vehicle-update branch. First peep in queue could easily be wrong I will have a look in a bit.

duncanspumpkin commented:
Var_114 is used for a few different things the 3 highest bits are number of eights of the ride that are covered. So a fully covered ride is 7 and a fully uncovered ride is 0. Hmm could label it and also put a note in saying its only the top 3 bits.

duncanspumpkin commented:
And yes it does look like its last peep in queue and not first as I originally wrote it.

clamburger commented:
Ah, I see now: the union containing inversions, holes and var_114 all refers to the same byte in memory, and the lower 5 bits are used for both inversions and holes (leaving the upper 3 bits for var_114). I'm still getting my head around how the memory is managed.

duncanspumpkin commented:
Forgot to mention some of the other things I was thinking of. 0x00982390 `RCT2_ADDRESS_NAUSEA_THRESHOLDS` turn that into a static array of the values. If you can't easily get the values I will do it at a later date. Could you also possibly make a new enum of the flags and then use them for every call to the function.

clamburger commented:
I've now made the requested changes.

duncanspumpkin commented:
Just checking but you changed it from three flags to two what was the other flag used for?

clamburger commented:
Flags 2 and 4 were always used together; although most of the function used flag 4 to check if the peep was at the ride, there was a [single instance](https://github.com/OpenRCT2/OpenRCT2/pull/2048/files#diff-aaf0461da04a85a49ced56a996781feaL7948) where flag 2 was checked instead.

Since it seemed that flags 2 and 4 did the same thing and are never used independently of each other, I replaced with single flag 2 check with a flag 4 check and updated any calls to the function that used flags 2+4 with just flag 4.

duncanspumpkin commented:
I just want to check one last thing with this then I will merge it
